kernel/src/process_checker.rs

```diff
@@ -58,40 +60,81 @@ impl<'a> AppCredentialsChecker<'a> for () {
     }
 }
 
-/// Return whether there is a currently running process that has
-/// the same application identifier as `process` OR the same short
-/// ID as `process`. This means that if `process` is currently
-/// running, `has_unique_identifier` returns false.
-pub fn has_unique_identifiers<AU: AppUniqueness>(
+/// Return whether `process` can run given the identifiers, version
+/// numbers, and execution state of other processes. A process is
+/// runnable if:
+///
+///   - Its Application Identifier and Short ID are different from
+///   all other processes, or
+///   - Other processes that share an Application Identifier or Short
+///   ID with it have a lower version number and are not running, or
+///   - Other processes that shared
```

Comment:
?

Reply:
Looks like it was truncated. Filled it in.